portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -39,27 +40,42 @@
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    String defaultCommentLimit = "5";
+    int commentLimit = Integer.parseInt(getParameter(request, "numShown", defaultCommentLimit));
+
+    String defaultPageNumber = "0";
+    int pageNumber = Integer.parseInt(getParameter(request, "pageNumber", defaultPageNumber));
 
+    String defaultSortMethod = "DESCENDING";
+    String sortMethod = getParameter(request, "sortMethod", defaultSortMethod);
+
+    Query query = new Query("Comment");
+    
+    if (sortMethod.equals(defaultSortMethod)) {
+      query.addSort("timestamp", SortDirection.DESCENDING);
+    } else {
```

Comment:
so if I call /data?sortMethod=DeScEnDiNg would it sort in ascending order?  How about if I mis-spell it?  That could make for a bad user experience.  Just make this parameter a boolean with a default value if not provided.

Reply:
Noted and done.